### PR TITLE
Signup reskin: Add the side section for the domain step

### PR DESCRIPTION
**Pull request by niranjan-uma-shankar**
#### Changes proposed in this Pull Request

This PR is towards implementing the reskinned signup experiment. For more context on the experiment, read pbxNRc-IO-p2.

NOTE:

This PR implements only the sidebar section of the domain step. The other changes will be implemented in subsequent PRs.

Here's the [Figma file](https://www.figma.com/file/gnYBUOBthTMckJRJr3pZmk/Control-Experiments?node-id=1114%3A536) for reference.

For the record:
Parent PR: https://github.com/Automattic/wp-calypso/pull/51483 and root PR: https://github.com/Automattic/wp-calypso/pull/50512

Screenshot 2021-03-31 at 4 04 45 PM

#### Testing instructions

* Assign yourself to treatment from Abacus, and visit /start.
* Proceed to the domains step and verify that the "sidebar" info section in the domains step matches the [Figma design](https://www.figma.com/file/gnYBUOBthTMckJRJr3pZmk/Control-Experiments?node-id=1114%3A536).
* Clicking on the CTA "View plans" should take to the plans step screen. The "Start with free" button should be hidden in that page.
* Clicking on the CTA "Use a domain I own" should take to the mappings/transfer page
* Start over and verify that control group experience is unaffected

### CODE REVIEW

File: client/signup/steps/domains/index.jsx [hunk 1]
```diff
@@ -602,14 +652,24 @@ class DomainsStep extends React.Component {
...
 
 		if ( isAllDomains ) {
 			return translate( 'Find the domain that defines you' );
 		}
 
 		if ( isReskinned ) {
-			return translate( "Enter your site's name or some descriptive keywords to get started" );
+			return (
+				! stepSectionName &&
+				translate( "Enter your site's name or some descriptive keywords to get started" )
```
Comment (a8ci18n) [inline client/signup/steps/domains/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **27** times:
`translate( 'Enter your site's name or some keywords that describe it to get started.' )` *ES Score: 6*

**Comment by taggon** [comment]
The page looks not good on mobile devices. Is that intended?
Create_a_site_—_WordPress_com

**Comment by niranjan-uma-shankar** [comment]
> The page looks not good on mobile devices. Is that intended?

I implemented the mobile device compatibility in a separate PR that is further down the stack: https://github.com/Automattic/wp-calypso/pull/51611. So, for this PR, please ignore the mobile views.

**Review by taggon — APPROVED** [review]
@niranjan-uma-shankar Okay, then, everything else looks good.

**Pull request merged**